com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -442,14 +443,15 @@ define([
                 $scope.errors.newStartDate = [];
                 $scope.errors.newEndDate = [];
 
-                $scope.validateDates($scope.edit_data.new_role_id.newStartDate, $scope.edit_data.new_role_id.newEndDate, function (error, field) {
-                    errors++;
-                    if (field.indexOf('start_date') > -1) {
-                        $scope.errors.newStartDate.push(error);
-                    }
-                    if (field.indexOf('end_date') > -1) {
-                        $scope.errors.newEndDate.push(error);
-                    }
+                var contract = me.contractsData[$scope.edit_data.new_role_id.job_contract_id];
```

Comment:
You have two variations of this:
```js
var contract = me.contractsData[$scope.edit_data.new_role_id.job_contract_id];
var contract = me.contractsData[data.contract.$viewValue];
```
what about to avoid duplication (in terms of *how* you get the contract data) and encapsulating it into a "private" function, something like
```js
/**
 *
 */
function contractData(contractId) {
    return me.contractsData[contractId];
}

// Usage
var contract = contractData($scope.edit_data.new_role_id.job_contract_id);
var contract = contractData(data.contract.$viewValue);

```